pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -744,6 +744,9 @@ type Karpenter struct {
 	// CreateServiceAccount create a service account or not.
 	// +optional
 	CreateServiceAccount *bool `json:"createServiceAccount,omitempty"`
+	// DefaultInstanceProfile override the default IAM instance profile
+	// +optional
+	DefaultInstanceProfile string `json:"defaultInstanceProfile,omitempty"`
```

Comment:
Please make this of type `*string` and then get the value as `aws.StringValue()`. The reason for that is that we truly would like to have this as an optional field. Which means that it's either set, is set to empty explicitly or set to nil which will completely remove it from parsed JSON data.

Reply:
Thanks, I did wonder about this but didn't see any immediate examples of `*string` hence what I went with. Will get this corrected.

Reply:
Or `ContainerRuntime`.